proxywasm/v1/shared.go

```diff
@@ -155,15 +159,16 @@ func ProxyGetSharedData(instance common.WasmInstance, keyPtr int32, keySize int3
 		return res.Int32()
 	}
 
-	err = instance.PutUint32(uint64(casPtr), uint32(cas))
+	err = instance.PutUint32(uint64(casPtr), cas)
 	if err != nil {
 		return WasmResultInvalidMemoryAccess.Int32()
 	}
 
 	return WasmResultOk.Int32()
 }
 
-func ProxySetSharedData(instance common.WasmInstance, keyPtr int32, keySize int32, valuePtr int32, valueSize int32, cas int32) int32 {
+func (a *ABIContext) ProxySetSharedData(keyPtr int32, keySize int32, valuePtr int32, valueSize int32, cas int32) int32 {
```

Comment:
the main change of this PR is moving the wasm instance to field instead of passed on the stack. This allows all the parameters and results to be compatible with webassembly types, which then allows compatability with other wasm runtimes which won't know to ignore the instance parameter.